analyzer/probes.go

```diff
@@ -46,21 +46,21 @@ func NewTopologyProbeBundleFromConfig(g *graph.Graph) (*probe.Bundle, error) {
 			continue
 		}
 
-		type newProbeFuncType func(*graph.Graph) (*k8s.Probe, error)
-		newProbeFuncMap := map[string]newProbeFuncType{
-			"k8s":   k8s.NewK8sProbe,
-			"istio": istio.NewIstioProbe,
-		}
+		var err error
 
-		if newProbeFunc, ok := newProbeFuncMap[t]; !ok {
+		switch t {
+		case "k8s":
+			probes[t], err = k8s.NewK8sProbe(g)
+		case "istio":
+			probes[t], err = istio.NewIstioProbe(g)
+		default:
 			logging.GetLogger().Errorf("unknown probe type: %s", t)
-		} else {
-			var err error
-			probes[t], err = newProbeFunc(g)
-			if err != nil {
-				logging.GetLogger().Errorf("failed to initialize probe %s: %s", t, err)
-				return nil, err
-			}
+			continue
+		}
+
+		if err != nil {
+			logging.GetLogger().Errorf("failed to initialize probe %s: %s", t, err)
+			return nil, err
```

Comment:
The user explicitely enabled the probe. If we failed to initialize the probe (which is different from starting the probe), we should fail.

Reply:
I basically did what the previous version was doing and that what we do for agent probes too. I think that OK since that is initialization of the probes. If some of them need resources that don't exist yet then they need to return no error and loop waiting for resources to be available. Neutron for example does this. Otherwise that real error and need to be reported directly stopping Skydive.

Reply:
gotcha! thanks!